lib/_http_agent.js

```diff
@@ -93,6 +94,12 @@ function Agent(options) {
     throw new ERR_INVALID_OPT_VALUE('scheduling', this.scheduling);
   }
 
+  if (this.options.maxTotalSockets !== undefined) {
+    validateNumber(this.options.maxTotalSockets, 'maxTotalSockets');
+    this.maxTotalSockets = this.options.maxTotalSockets;
+    this.totalSocketCount = 0;
```

Comment:
These two instance properties should be set all the time to keep the object shapes the same, no matter the configuration.

Reply:
fixed